raster/src/main/scala/geotrellis/raster/ResampleTarget.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster
+
+import geotrellis.raster.reproject.Reproject
+
+import spire.math.Integral
+import spire.implicits._
+
+/** Represents a strategy/target for resampling */
+sealed trait ResampleTarget {
```

Comment:
Mb these are my language problems, but `ResampleGird` is a bit more understandable for me rather than `ResampleTarget`. Is not `Target` a too generic thing? Like `Data` when we were designing `Paths` / `DataPaths` for `RasterSources`. 

I am not defending `ResampleGrid` name, I am not fan of it either.

Reply:
Target might be too generic, but I guess i'm still a bit stuck on the fact that it isn't a ResampleGrid so much as a strategy/target by which a grid will be summoned/constructed

Reply:
@moradology totally agree, it is not `ResampleGrid`, I don't have a better naming suggestion.